datafusion/common/src/column.rs

```diff
@@ -112,12 +115,21 @@ impl Column {
 
     /// Serialize column into a quoted flat name string
     pub fn quoted_flat_name(&self) -> String {
-        // TODO: quote identifiers only when special characters present
-        // see: https://github.com/apache/arrow-datafusion/issues/5523
         match &self.relation {
             Some(r) => {
-                format!("{}.{}", r.to_quoted_string(), quote_identifier(&self.name))
+                lazy_static! {
+                    static ref CAPTURE_VALID_RE: Regex =
```

Comment:
I spent some more time testing this PR out locally and I found:
1. `quoted_identifier` didn't seem to have many tests
2. I think we still need to quote any identifier that has capitals (as it would be normalized to lowercase)

I'll make a PR shortly with an alternate proposal